### PR TITLE
PREAPPS-2943: Tags | Create and edit tags

### PR DESCRIPTION
**Pull request by avadhutprabhudesai**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -384,6 +385,19 @@ export class ZimbraBatchClient {
 			body: denormalize(CreateSignatureRequest)(options)
 		});
 
+	public createTag = (_options: CreateTagOptions) => {
```
Comment (silentsakky) [inline src/batch-client/index.ts]: `({ name, color }: CreateTagOptions)`

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -384,6 +385,19 @@ export class ZimbraBatchClient {
...
+		const { name, color } = _options;
+		return this.jsonRequest({
+			name: 'CreateTag',
+			body: {
+				tag: {
+					name,
```
Comment (silentsakky) [inline src/batch-client/index.ts]: do we need destructuring and separate keys here?
```
tag: {
    ...options
}
```

File: src/schema/schema.graphql [hunk 3]
```diff
@@ -1922,6 +1922,14 @@ input FolderActionInput {
 	zimbraId: ID
 }
 
+input TagActionRequestInput {
```
Comment (silentsakky) [inline src/schema/schema.graphql]: shouldn't be this same as whatever is passed in itemAction, can we make both methods generic and have consistency?

File: src/schema/schema.ts [hunk 4]
```diff
@@ -438,7 +439,13 @@ export function createZimbraSchema(
 				setRecoveryAccount: (_, variables) =>
 					client.setRecoveryAccount(variables as SetRecoveryAccountOptions),
 				modifyWhiteBlackList: (_, { whiteBlackList }) =>
-					client.modifyWhiteBlackList(whiteBlackList as WhiteBlackListInput)
+					client.modifyWhiteBlackList(whiteBlackList as WhiteBlackListInput),
+				createTag: (_, { name, color }) =>
```
Comment (silentsakky) [inline src/schema/schema.ts]:
```
createTag: (_, variables) =>
					client.createTag(variables as CreateTagOptions),
```

File: src/schema/schema.graphql (hunk 3 above)
Comment (avadhutprabhudesai) [reply src/schema/schema.graphql]: Done.

File: src/batch-client/index.ts (hunk 2 above)
Comment (avadhutprabhudesai) [reply src/batch-client/index.ts]: Done

File: src/batch-client/index.ts (hunk 1 above)
Comment (avadhutprabhudesai) [reply src/batch-client/index.ts]: Obsolete comment.

File: src/schema/schema.ts (hunk 4 above)
Comment (avadhutprabhudesai) [reply src/schema/schema.ts]: Done.

File: src/batch-client/index.ts [hunk 5]
```diff
@@ -384,6 +385,17 @@ export class ZimbraBatchClient {
 			body: denormalize(CreateSignatureRequest)(options)
 		});
 
+	public createTag = (tag: CreateTagInput) => {
```
Comment (itsmeprashant) [inline src/batch-client/index.ts]: no need of extra `{`

Comment (avadhutprabhudesai) [reply src/batch-client/index.ts]: Done

**Pull request merged**